src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1735,12 +1735,7 @@ private static void GetStateMachineSlotDebugInfo(
 
             initializersBody ??= GetSynthesizedEmptyBody(method);
 
-            if (method is SynthesizedPrimaryConstructor primaryCtor && method.ContainingType.IsStructType())
-            {
-                body = BoundBlock.SynthesizedNoLocals(primaryCtor.GetSyntax());
-                nullableInitialState = getInitializerState(body);
-            }
-            else if (method is SourceMemberMethodSymbol sourceMethod)
```

Comment:
This path was causing us to miss nullable analyzing the struct primary constructor.